core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -64,57 +64,59 @@ export class AdminFeaturesTabComponent implements OnInit {
       inputRegex?: RegExp;
     }
   } = {
-    [PlatformParameterFilterType.ServerMode]: {
-      displayName: 'Server Mode',
-      options: AdminFeaturesTabConstants.ALLOWED_SERVER_MODES,
-      operators: ['='],
-      optionFilter: (feature: PlatformParameter, option: string): boolean => {
-        switch (feature.featureStage) {
-          case FeatureStage.DEV:
-            return option === 'dev';
-          case FeatureStage.TEST:
-            return option === 'dev' || option === 'test';
-          case FeatureStage.PROD:
-            return true;
-          default:
-            return false;
+      [PlatformParameterFilterType.ServerMode]: {
```

Comment:
is this a required/mandatory change?

Reply:
Yep this is required by eslint now.

Reply:
I don't understand, why is it required? Could you please point to the documentation/discussion? It feels like the sort of thing that should be configurable.

Otherwise the whole thing feels inconsistent with our "2-space indentation for everything" rule.

Reply:
I think the problem is that we use the eslint `indent` rule instead of using `typescript-esllint/indent`, but the problem is that the developers of  `typescript-esllint/indent` do not recommend not using it as it is not implemented properly (https://github.com/typescript-eslint/typescript-eslint/issues/1824), I can add an ignore here if we want to strictly keep 2 spaces.

Reply:
But where is the part of the documentation where `eslint` itself requires this behaviour (and it can't be configured)?

Reply:
I don't think it is mentioned anywhere, it is probably a bug in the indent implementation that doesn't expect TS constructs (for that there is `typescript-esllint/indent` but that is broken in other ways) and for some reason assume this should be indented by 4 spaces. If I remove the typing info it will change and expect 2 spaces.

Reply:
OK, can you file an issue on their repo, and on ours (with full details) to fix it once this issue is fixed on their end?

In general, anything that isn't in line with what is expected should always be filed and tracked somewhere.

Reply:
From the nature of the issue, they will not fix it, because it is caused by TypeScript and they do no intent to fix indent and recommend replacing it with code formatter like Prettier, I'm fine with filing an open-ended issue in our repo explaining the problem and proposing the possible solutions.

Reply:
OK, sgtm. Could you please file the issue and link it here, and then I will approve the PR.

Reply:
#14605